reply.py

```diff
@@ -25,17 +25,23 @@ async def run(self):
 	async def process_notification(self, notification):
 		acct = "@" + notification['account']['acct']  # get the account's @
 		post_id = notification['status']['id']
-		context = await self.pleroma.status_context(post_id)
 
-		# check if we've already been participating in this thread
-		if self.check_thread_length(context):
-			return
-
-		content = self.extract_toot(notification['status']['content'])
-		if content in {'pin', 'unpin'}:
-			await self.process_command(context, notification, content)
+		try:
+			context = await self.pleroma.status_context(post_id)
+		except pleroma.BadResponse as exc:
+		    # just try again, hopefully we don't exhaust the call stack lol
```

Comment:
It's pretty likely that a 500 error is not intermittent. Pass a retry count to process_notification, and log the 500 exception instead if it fails three times in a row. Sleep 2**(retry_count) in between each try. (use `await anyio.sleep(n)`)